### DISCUSSION TITLE
Fail setWebhook

### DISCUSSION THREAD

**Discussion by roman-rachkov** [def-studio/telegraph#116] [Q&A] [answered]
Hello. I try use command php artisan telegraph set-webhook, but im recive error
```
Failed to register webhook
{"ok":false,"error_code":400,"description":"Bad Request: bad webhook: IP address 127.0.0.1 is reserved"}

```
I think this becouse im use docker compose configuration on my vds. But how fix it no idea.

Empty laravel 9.19 project, just install sanctum, orchid and telegraph.

thx

**Comment by fabio-ivona** [comment, marked as answer]
Hi, your url is generated from `telagraph.webhook` route:

```php
public function registerWebhook(): Telegraph
{
    $telegraph = clone $this;

    $url = route('telegraph.webhook', $telegraph->getBot());

    // ...
```

so your app should generate a valid url for that route (the domain should be set in your env APP_URL value)

**Reply by roman-rachkov** [reply]
Im stupid. thx

**Comment by ah-rahimi** [comment]
How To test webhook in local 
![image](https://user-images.githubusercontent.com/87445571/180248934-4c2c496c-a811-4fed-841e-14f362b2edfd.png)
in my local webhook have error

**Reply by ah-rahimi** [reply]
@fabio-ivona

**Reply by fabio-ivona** [reply]
there's not an easy way to use telegram bots in a local environment, Telegram requires webhooks to have a secure and public url, so https and reachable from telegram servers

**Comment by horaceho** [comment]
I setup webhook in local environment using `valet share`:

1. `valet secure` the Laravel project
2. `valet share` to forward the local site via `ngrok`
3. Copy the forwarding URL in the `valet share` console, for example: `https://xxxx-101-101-101-101.ngrok.io`
4. Paste the URL in `.env` `APP_URL=`
5. In BotFather, `/setdomain` to `xxxx-101-101-101-101.ngrok.io`
6. Run `php artisan telegraph:set-webhook` to enable webhook
